moveit_simple/include/moveit_simple/trajectory_processing.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2019 Plus One Robotics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <moveit_simple/exceptions.h>
+#include <moveit/trajectory_processing/time_optimal_trajectory_generation.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/robot_trajectory/robot_trajectory.h>
+
+#ifndef TRAJECTORY_PROCESSING_H
+#define TRAJECTORY_PROCESSING_H
+namespace moveit_simple
+{
+namespace trajectory_processing
+{
+struct TrajectoryValidationResult
+{
+  enum { Success,
+         InvalidPosition,
+         InvalidVelocity,
+         InvalidAcceleration,
+         InvalidTimestamp} value;
+  std::string error_message;
+};
+TrajectoryValidationResult validateTrajectory(robot_model::RobotModelConstPtr robot_model,
+                                              const trajectory_msgs::JointTrajectory& trajectory)
+{
+  TrajectoryValidationResult result;
+  const std::vector<std::string>& joint_names = trajectory.joint_names;
+  trajectory_msgs::JointTrajectoryPoint previous_point;
+  for (size_t i = 0; i < trajectory.points.size(); ++i)
+  {
+    // validate waypoint bounds
+    const auto& point = trajectory.points[i];
+    for (size_t j = 0; j < joint_names.size(); ++j)
+    {
+      const auto& joint_model = robot_model->getJointModel(joint_names[j]);
+      const auto& bounds = joint_model->getVariableBounds()[0];
+      // validate position bounds and throw exception since this can't be fixed easily
+      if (bounds.position_bounded_ && !joint_model->satisfiesPositionBounds(&point.positions[j]))
+      {
+        result.value = TrajectoryValidationResult::InvalidPosition;
+        result.error_message = "Trajectory contains waypoints that don't satisfy position bounds";
```

Comment:
I would add more diagnostic information (eg. which joint, which index, what the invalid value was and what the bound was)